meta-iotqa/lib/oeqa/runtime/multimedia/vaapi/test_vaapi_encode_decode.py

```diff
@@ -0,0 +1,198 @@
+import os
+from oeqa.oetest import oeRuntimeTest
+
+'''
+This test suit tests VAAPI Decoding and Encoding
+'''
+
+
+class VAAPITest(oeRuntimeTest):
+
+    '''
+    The video files have been created using gstreamer and videotestsrc as input
+
+    '''
+
+    h264_encoded_file = "h264_enc.mp4"
+    # mpeg2_encoded_file = 'mpeg2_enc.ts'
+    # vc1_encoded_file = 'vc1_enc.wmv'
+    mjpeg_encoded_file = 'mjpeg_enc.avi'
+    vp8_encoded_file = 'vp8_enc.webm'
+    # vp9_encoded_file = 'vp9_enc.webm'
+
+    h264_encoded_file_target = '/tmp/%s' % h264_encoded_file
+    # mpeg2_encoded_file_target = '/tmp/%s' % mpeg2_encoded_file
+    # vc1_encoded_file_target = '/tmp/%s' % vc1_encoded_file
+    mjpeg_encoded_file_target = '/tmp/%s' % mjpeg_encoded_file
+    vp8_encoded_file_target = '/tmp/%s' % vp8_encoded_file
+    # vp9_encoded_file_target = '/tmp/%s' % vp9_encoded_file
+
+
+    available_vaapi_elements = ""
+
+    def setUp(self):
+        # Copy all necessary files for test to the target device.
+        self.target.copy_to(
+            os.path.join(
+                os.path.dirname(__file__),
+                'files',
+                VAAPITest.h264_encoded_file),
+            VAAPITest.h264_encoded_file_target)
+        '''
+        self.target.copy_to(
+            os.path.join(
+                os.path.dirname(__file__),
+                'files',
+                VAAPITest.mpeg2_encoded_file),
+            VAAPITest.mpeg2_encoded_file_target)
+
+        self.target.copy_to(
+            os.path.join(
+                os.path.dirname(__file__),
+                'files',
+                VAAPITest.vc1_encoded_file),
+            VAAPITest.vc1_encoded_file_target)
+        '''
+        self.target.copy_to(
+            os.path.join(
+                os.path.dirname(__file__),
+                'files',
+                VAAPITest.mjpeg_encoded_file),
+            VAAPITest.mjpeg_encoded_file_target)
+
+        self.target.copy_to(
+            os.path.join(
+                os.path.dirname(__file__),
+                'files',
+                VAAPITest.vp8_encoded_file),
+            VAAPITest.vp8_encoded_file_target)
+
+        # get supported vaapi encoders and decoders
+
+        (status, output) = self.target.run(" gst-inspect-1.0 | grep vaapi")
+        output = output.lower()
+        self.available_vaapi_elements = output
```

Comment:
all of the test functions expect ```vaapisink``` which is not checked anywhere. please add a sanity check for it.